XMLParser/XMLParser.cpp

```diff
@@ -0,0 +1,361 @@
+// This is an open source non-commercial project. Dear PVS-Studio, please check it.
+
+// PVS-Studio Static Code Analyzer for C, C++ and C#: http://www.viva64.com
+
+#include "pch.h"
+#include "framework.h"
+#include "XMLParser.h"
+
+bool XMLParser::ParseFile(const std::string& filename)
+{
+	struct Characters
+	{
+		const char tag1 = '<';
+		const char tag2 = '>';
+		const char slash = '/';
+		const std::string exclamation = "<!";
+		const std::string question = "<?";
+		const std::string end_tag_symbols = "</";
+	};
+	std::filesystem::path file_path = filename;
+	if (!FileCheck(file_path))
+	{
+		return false;
+	}
+
+	std::ifstream file{ filename };
+	if (!file.is_open())
+	{
+		//std::cout << "Can't open file.\n";
+		return false;
+	}
+
+	std::string line;
+	std::unique_ptr<Characters> symbol = std::make_unique<Characters>();
+	bool tag_flag = false;
+	bool first_flag = true;
+	std::shared_ptr<XMLNode> current_node = nullptr;
```

Comment:
We don't need to nullify smart pointers.

Reply:
We don't need to, but it's better for code readability.